muzzle/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/AgentTooling.java

```diff
@@ -16,11 +16,14 @@
  */
 public final class AgentTooling {
 
+  private static final AgentLocationStrategy LOCATION_STRATEGY =
+      locationStrategy(getBootstrapProxy());
+
   private static final AgentBuilder.PoolStrategy POOL_STRATEGY =
-      new AgentCachingPoolStrategy(locationStrategy(getBootstrapProxy()));
+      new AgentCachingPoolStrategy(LOCATION_STRATEGY);
 
   public static AgentLocationStrategy locationStrategy() {
-    return locationStrategy(null);
+    return LOCATION_STRATEGY;
```

Comment:
Just to confirm the fix is to not pass `null` in this codepath?

Reply:
Yes. The problem is probably that wildfly class loader doesn't unconditionally delegate resource lookups to system loader and isn't able to find resources that are in agent jar. Previously it worked because in `AgentLocationStrategy` we added `ClassFileLocator` for all parent class loaders, which included system class loader, where the agent is initially loaded. To make resources from agent jar visible we need to add a `ClassFileLocator` with bootstrap proxy.